src/alias_scraper.py

```diff
@@ -0,0 +1,48 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup
+import json
+import re
+
+
+def get_players_list():
+  url = 'https://warcraft3.info/stats/player'
+  page = urlopen(url)
+  soup = BeautifulSoup(page, 'html.parser')
+  player_urls = []
+  for a in soup.find_all('a', {"class": "stats_player_field"}):
+    link = a.get('href')
+    player_urls.append(link)
+  print('Found {} urls'.format(len(player_urls)))
+  return player_urls
+
+def get_players_alts(soup):
+  alts = []
+  for a in soup.find_all('a', {"title": re.compile(r'^Visit.*profile on Battle.net')}):
+    if a is None:
+      continue
+    link = a.get('href')
+    player_name_pat = re.compile(r'.*PlayerName=(.*)')
+    player_name = player_name_pat.match(link).group(1)
+    alts.append(player_name)
+  return alts
+
+
+player_urls = get_players_list()
+aliases = {}
+
+print('Finding aliases...')
+for url in player_urls:
```

Comment:
This takes a really long time to run. Some options:
1) It might be more efficient to just scrape https://warcraft3.info/stats/bnet_ranking/<gateway> 

2) Make it multi-threaded:
``
def parse_player_page(q):
  queue_full = True
  while queue_full:
    try:
      url = q.get(False)
      print(url)
      page = urlopen(url)
      soup = BeautifulSoup(page, 'html.parser')
      pat = re.compile(r'.*/(.*)$')
      player = pat.match(url).group(1)
      alts = get_players_alts(soup)
      for alt in alts:
        aliases[alt] = player
    except queue.Empty:
      queue_full = False


player_urls = get_players_list()
aliases = {}

q = queue.Queue()
for url in player_urls:
    q.put(url)

print('Finding aliases...')
thread_count = 10  #if you make this too high you might get blocked for potential DDOS
for i in range(thread_count):
    t = threading.Thread(target=parse_player_page, args = (q,))
    t.start()
``

Reply:
I looked at that page before but I was worried paginating it would be a pain, however it looks like the data is being exposed on an endpoint that we could just consume from instead: https://warcraft3.info/stats/bnet_data?server=Northrend

```
{
      "id": "9981040",
      "delta": "",
      "rank": "4",
      "level": "48",
      "exp": "21477",
      "wins": "2590",
      "losses": "378",
      "winrate": "87%",
      "server": "Northrend",
      "name": "<a href=\"http:\/\/classic.battle.net\/war3\/ladder\/w3xp-player-profile.aspx?Gateway=Northrend&PlayerName=RomanticHuman\" title=\"Visit RomanticHuman's profile on Battle.net\" target=\"_blank\">RomanticHuman<\/a>",
      "created_at": "2019-03-05 22:10:30",
      "aka": "<i class=\"flag-icon flag-icon-fr\"><\/i><i class=\"race race-human\"><\/i> <a href=\"..\/..\/stats\/player\/ToD\" title=\"See ToD's stats\">ToD<\/a>",
      "daily_history": {
        "id": "768751",
        "rank": "4",
        "level": "48",
        "exp": "21477",
        "wins": "2590",
        "losses": "378",
        "winrate": "87",
        "server": "Northrend",
        "name": "RomanticHuman",
        "created_at": "2019-02-25 23:55:34",
        "history_type": "daily"
      }
},
```

In this case it might not even be necessary to scrape the data, we could just make a request to this endpoint in real time.

Reply:
That would be nice!

Reply:
See latest commit 1ba8ce0fad3d176bc7b3ef0c188e25cd84238919